CONTRIBUTING.md

```diff
@@ -0,0 +1,269 @@
+
+- **Enhancements**
+    + Any implementations of pre-filed enhancement requests should clearly link to the original issue.
+    + Any new enhancements should be documented as if filing a new enhancement request using the template above.
+- **Bug Fixes**
+    + Any fixes for pre-filed issues should clearly link to the original issue.
+    + Any newly found issues should be documented as if filing a new issue using the template above.
+
+### Exceptions ###
+
+Exceptions to these criteria will be handled on a case-by-case basis, and should be discussed via the Issues tab.
+
+## Code Structure ##
+
+The organization of the rocWMMA library is explained in detail in the [Programmers Guide](https://github.com/ROCm/rocWMMA/blob/develop/docs/Programmers_Guide.rst)
+
+## Coding Style ##
+
+This project follows the [CPP Core Guidelines](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md)
+with few modifications or additions noted below. All pull-requests should in good faith attempt to follow the guidelines stated therein,
+but we recognize that the content is lengthy. Below we list our primary concerns when reviewing pull-requests.
+
+### Interface ###
+
+- Library code should use C++17.
+- Our minimum supported compiler is hipcc 4.4.
+- Avoid CamelCase.
+    * This rule applies specifically to publicly visible APIs, but is also encouraged (not mandated) for internal code
+
+### Philosophy ###
+
+-  [P.2](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rp-Cplusplus)
+   Write in ISO Standard C++14 (especially to support Windows, Linux and
+   MacOS platforms)
+-  [P.5](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rp-compile-time)
+   Prefer compile-time checking to run-time checking
+
+### Implementation ###
+
+-  [SF.1](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rs-file-suffix)
+   Use a ``.cpp`` suffix for code files and an ``.hpp`` suffix for
+   interface files if your project doesn't already follow another
+   convention
+-  [SF.5](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rs-consistency)
+   A ``.cpp`` file must include the ``.hpp`` file(s) that defines its
+   interface
+-  [SF.7](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rs-using-directive)
+   Don't put a global ``using``-directive in a header file
+-  [SF.8](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rs-guards)
+   Use ``#include`` guards for all ``.hpp`` files
+-  [SF.21](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rs-unnamed)
+   Don't use an unnamed (anonymous) ``namespace`` in a header
+-  [SL.10](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rsl-arrays)
+   Prefer using ``std::array`` or ``std::vector`` instead of a C array
```

Comment:
This should be removed specifically for hipRTC. We don't want to encourage std objects in the main library.